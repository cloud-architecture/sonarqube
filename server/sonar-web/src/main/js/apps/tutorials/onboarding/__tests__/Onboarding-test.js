/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// @flow
import React from 'react';
import { shallow, mount } from 'enzyme';
import Onboarding from '../Onboarding';
import { click, doAsync } from '../../../../helpers/testUtils';

jest.mock('../../../../api/users', () => ({
  skipOnboarding: () => Promise.resolve()
}));

const currentUser = { login: 'admin', isLoggedIn: true };

it('guides for on-premise', () => {
  const wrapper = shallow(
    <Onboarding
      currentUser={currentUser}
      onSkip={jest.fn()}
      organizationsEnabled={false}
      sonarCloud={false}
    />
  );
  expect(wrapper).toMatchSnapshot();

  // $FlowFixMe
  wrapper.instance().handleTokenDone('abcd1234');
  wrapper.update();
  expect(wrapper).toMatchSnapshot();
});

it('guides for sonarcloud', () => {
  const wrapper = shallow(
    <Onboarding
      currentUser={currentUser}
      onSkip={jest.fn()}
      organizationsEnabled={true}
      sonarCloud={true}
    />
  );
  expect(wrapper).toMatchSnapshot();

  // $FlowFixMe
  wrapper.instance().handleOrganizationDone('my-org');
  wrapper.update();
  expect(wrapper).toMatchSnapshot();

  // $FlowFixMe
  wrapper.instance().handleTokenDone('abcd1234');
  wrapper.update();
  expect(wrapper).toMatchSnapshot();
});

it('skips', () => {
  const onSkip = jest.fn();
  const wrapper = mount(
    <Onboarding
      currentUser={currentUser}
      onSkip={onSkip}
      organizationsEnabled={false}
      sonarCloud={false}
    />
  );
  click(wrapper.find('.js-skip'));
  return doAsync(() => {
    expect(onSkip).toBeCalled();
  });
});
