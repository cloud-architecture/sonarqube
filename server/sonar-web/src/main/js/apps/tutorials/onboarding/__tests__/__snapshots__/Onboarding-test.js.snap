// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`guides for on-premise 1`] = `
<div
  className="page page-limited"
>
  <header
    className="page-header"
  >
    <h1
      className="page-title"
    >
      onboarding.header
    </h1>
    <div
      className="page-actions"
    >
      <a
        className="js-skip text-muted"
        href="#"
        onClick={[Function]}
      >
        tutorials.skip
      </a>
    </div>
    <div
      className="page-description"
    >
      onboarding.header.description
    </div>
  </header>
  <TokenStep
    onContinue={[Function]}
    open={true}
    stepNumber={1}
  />
  <AnalysisStep
    onFinish={[Function]}
    onReset={[Function]}
    open={false}
    sonarCloud={false}
    stepNumber={2}
  />
</div>
`;

exports[`guides for on-premise 2`] = `
<div
  className="page page-limited"
>
  <header
    className="page-header"
  >
    <h1
      className="page-title"
    >
      onboarding.header
    </h1>
    <div
      className="page-actions"
    >
      <a
        className="js-skip text-muted"
        href="#"
        onClick={[Function]}
      >
        tutorials.skip
      </a>
    </div>
    <div
      className="page-description"
    >
      onboarding.header.description
    </div>
  </header>
  <TokenStep
    onContinue={[Function]}
    open={false}
    stepNumber={1}
  />
  <AnalysisStep
    onFinish={[Function]}
    onReset={[Function]}
    open={true}
    sonarCloud={false}
    stepNumber={2}
    token="abcd1234"
  />
</div>
`;

exports[`guides for sonarcloud 1`] = `
<div
  className="page page-limited"
>
  <header
    className="page-header"
  >
    <h1
      className="page-title"
    >
      onboarding.header.sonarcloud
    </h1>
    <div
      className="page-actions"
    >
      <a
        className="js-skip text-muted"
        href="#"
        onClick={[Function]}
      >
        tutorials.skip
      </a>
    </div>
    <div
      className="page-description"
    >
      onboarding.header.description
    </div>
  </header>
  <OrganizationStep
    currentUser={
      Object {
        "isLoggedIn": true,
        "login": "admin",
      }
    }
    onContinue={[Function]}
    open={true}
    stepNumber={1}
  />
  <TokenStep
    onContinue={[Function]}
    open={false}
    stepNumber={2}
  />
  <AnalysisStep
    onFinish={[Function]}
    onReset={[Function]}
    open={false}
    sonarCloud={true}
    stepNumber={3}
  />
</div>
`;

exports[`guides for sonarcloud 2`] = `
<div
  className="page page-limited"
>
  <header
    className="page-header"
  >
    <h1
      className="page-title"
    >
      onboarding.header.sonarcloud
    </h1>
    <div
      className="page-actions"
    >
      <a
        className="js-skip text-muted"
        href="#"
        onClick={[Function]}
      >
        tutorials.skip
      </a>
    </div>
    <div
      className="page-description"
    >
      onboarding.header.description
    </div>
  </header>
  <OrganizationStep
    currentUser={
      Object {
        "isLoggedIn": true,
        "login": "admin",
      }
    }
    onContinue={[Function]}
    open={false}
    stepNumber={1}
  />
  <TokenStep
    onContinue={[Function]}
    open={true}
    stepNumber={2}
  />
  <AnalysisStep
    onFinish={[Function]}
    onReset={[Function]}
    open={false}
    organization="my-org"
    sonarCloud={true}
    stepNumber={3}
  />
</div>
`;

exports[`guides for sonarcloud 3`] = `
<div
  className="page page-limited"
>
  <header
    className="page-header"
  >
    <h1
      className="page-title"
    >
      onboarding.header.sonarcloud
    </h1>
    <div
      className="page-actions"
    >
      <a
        className="js-skip text-muted"
        href="#"
        onClick={[Function]}
      >
        tutorials.skip
      </a>
    </div>
    <div
      className="page-description"
    >
      onboarding.header.description
    </div>
  </header>
  <OrganizationStep
    currentUser={
      Object {
        "isLoggedIn": true,
        "login": "admin",
      }
    }
    onContinue={[Function]}
    open={false}
    stepNumber={1}
  />
  <TokenStep
    onContinue={[Function]}
    open={false}
    stepNumber={2}
  />
  <AnalysisStep
    onFinish={[Function]}
    onReset={[Function]}
    open={true}
    organization="my-org"
    sonarCloud={true}
    stepNumber={3}
    token="abcd1234"
  />
</div>
`;
